### PR TITLE
Refinements to the workflow #4714

### LINKED ISSUE #4714: Refinements to the workflow
I propose following changes to the workflow:

**Issue life cycle:**
* An issue in `s.toMerge` state should be verified by PM (indicated by the `s.mergeApproved` label) before it is merged. If the PM asks for further changes, the issue goes back to `s.ongoing`
Reason: we can't rely on just one peer reviewer to ensure the code quality.  
* Senior developers and above should not take lower priority issues when there are open higher priority issues. Committers are encouraged to do the same.
Reason: we have very limited man power. We should spend it on high priority issues. That's the whole purpose of having priorities.

**Team Lead responsibilities:**
* Ensure the Milestone plan should be ready at the beginning of the milestone.
Reason: Developers need more practice on planning and estimation. 

**Coding style:**
* All public members and non-trivial private members should have header comments.
Reason: Header comments help in ensuring members are well-defined. e.g. conformance to Single Responsibility Principle.

### PR DESCRIPTION
**Pull request by thyageshm**
fixes #4714

### CODE REVIEW

**Comment by thyageshm** [comment]
@damithc sir the pr is ready for your review :-)

File: devdocs/process.md [hunk 1]
```diff
@@ -112,6 +112,10 @@ This workflow is an adaptation of the [GitHub flow](https://guides.github.com/in
      is to continue until PR status changes to `s.toMerge`. After doing suggested
      changes, remember to add a comment to indicate the PR is ready for review again.
      e.g. `ready to review` or `changes done`
+
+   * As a final check, the PM will look through the changes and either suggest changes (back to `s.Ongoing`),
+     or apply the `s.mergeApproved` label to the PR. Rarely, there would be multiple jumps between `s.Ongoing` and `s.toMerge`
+     before the fix is applied the `s.mergeApproved` status.
```
Comment (damithc) [inline devdocs/process.md]: No need for the 2nd sentence.

File: devdocs/process.md [hunk 2]
```diff
@@ -131,12 +135,24 @@ Role: reviewer
...
     * Optionally, add a comment in the conversation thread to inform the author to refine the code.
   * To remove white space changes from being shown, append `?w=1` to url of the `/files` page of the pull request (the "Files changed" tab)
   * If the code is OK on all aspects,
     * Change issue status to `s.ToMerge`
 
+Role: PM
+
+  * Review the code for maintainability and style
+  * Ensure appropriate header comments and expected standards are followed
+    * the standards used in TEAMMATES are available at [Readme](../README.md) under the *Supplementary documents* section
```
Comment (damithc) [inline devdocs/process.md]: These things are reviewer's job. PM does `a final check on the code quality and style`.

**Comment by damithc** [comment]
I think some diagrams need to change too.

**Comment by thyageshm** [comment]
I have made the necessary changes. Moved the responsibility to check for standards to the reviewer

**Comment by damithc** [comment]
A few more things missing I think. Go through the comments in the issue and the PR to ensure all points are covered.

**Comment by thyageshm** [comment]
@damithc I have made the changes

**Comment by damithc** [comment]
The two diagrams need slight changes too.

**Comment by thyageshm** [comment]
sir, do you have the psd file or any of the raw files for me to use?

**Comment by damithc** [comment]
The sources are are in pptx format. The file is somewhere in the repo.

**Comment by thyageshm** [comment]
ok! found them :-)

**Comment by thyageshm** [comment]
Sir, for merging to master and pushing to the committer repo, should I get the reviewer or the dev to do it in the diagram? In our case, the dev usually does it if she has the access and the reviewer does it otherwise

**Comment by damithc** [comment]
Let's draw it to reflect the workflow for contributors. i.e. reviewer merges. 
Letting the dev merge is a shortcut we take when the dev is experienced enough.

**Comment by thyageshm** [comment]
@damithc sir, I have updated the two diagrams

**Comment by damithc** [comment]
In the issue life cycle, instead of `s.Accepted` (we don't have that any more), say `Prioritized`.
Instead of `s.Ongoing` on the issue side, say `Assigned`.
Can go ahead and merge after that.

**Pull request merged**